src/Query/Query.ts

```diff
@@ -161,7 +170,14 @@ export class Query implements IQuery {
         const { debugSettings } = getSettings();
         const tasksSorted = debugSettings.ignoreSortInstructions ? tasks : Sort.by(this.sorting, tasks);
         const tasksSortedLimited = tasksSorted.slice(0, this.limit);
-        return new TaskGroups(this.grouping, tasksSortedLimited);
+
+        const taskGroups = new TaskGroups(this.grouping, tasksSortedLimited);
+
+        if (this._taskGroupLimit !== undefined) {
+            taskGroups.applyTaskLimit(this._taskGroupLimit);
+        }
+
+        return taskGroups;
```

Comment:
@claremacrae I thought about including `this._taskGroupLimit` in `TaskGroups.construtor()` parameters, but then I would need to modify a lot of tests with calls like 
```js
const groups = new TaskGroups(grouping, inputs, undefined);
```

where the `undefined` looks horribly senseless and obliges to look in the description of the constructor, in other words it is not self-evident like the 2 other parameters.

The other option would be to make an optional parameter with a default value but as we discussed this may result in errors later. Pls let me know if you would prefer a way different from what I did.